packages/pigeon/README.md

```diff
@@ -1,13 +1,354 @@
 # Pigeon
 
+// TODO: how to frame this?
+// To get started, see the Quickstart Guide in the [Example README](./example/README.md).
+
 Pigeon is a code generator tool to make communication between Flutter and the
 host platform type-safe, easier, and faster.
 
-Pigeon removes the necessity to manage strings across multiple platforms and languages.
-It also improves efficiency over common method channel patterns. Most importantly though,
-it removes the need to write custom platform channel code, since pigeon generates it for you.
+Pigeon works by reading special files which by convention are placed in a separate
+folder, `/pigeons`, alongside your `/lib` folder. You can put all of your definitions
+in a single file (e.g., `/pigeons/messages.dart`), or you can divide your methods
+and classes into separate files as you wish. After reading these files, Pigeon
+generates:
+
+ * New Dart source files within your `/lib` folder at the specific paths you
+specify, and
+ * New native source files with matching code within your native projects at the
+ specific paths you specify
+
+Internally, the generated code uses `MethodChannel`s to communicate across Flutter's
+UI thread and the Platform thread where your host app initially launched. The value
+in Pigeon comes from automatically keeping this unpleasant boilerplate in sync and
+efficiently marshalling data between languages.
```

Comment:
This is an internal implementation detail, not something people generally need to know.

Reply:
FWIW, Our existing documentation does mention efficiency gains which personally made me wonder, "O RLY? Like what?"

Reply:
These changes don't describe the efficiency gains, so I'm not sure how that related to adding this paragraph.

If we want to add a technical discussion somewhere of how Pigeon improves over the common pattern of using string-keyed dictionaries to send multiple parameters we could, but that would be completely orthogonal to making it easier to understand how to use Pigeon (and IMO wouldn't be a great use of README space).